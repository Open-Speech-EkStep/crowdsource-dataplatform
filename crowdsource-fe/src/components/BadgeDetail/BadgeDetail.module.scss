.highlight {
  color: #ffcc35;
}

.tabs {
  background-color: var(--light-color);
  flex-wrap: nowrap;
  height: 4rem;
  -ms-overflow-style: none; /* IE and Edge */
  overflow-x: auto;
  scrollbar-width: none; /* Firefox */

  li {
    flex: 1;
    height: 100%;
  }
}

.tabLink {
  width: max-content;

  @include media-breakpoint-up(md) {
    width: 100%;
  }

  &:global(.active)::before {
    border: 2px solid var(--primary-color);
    border-radius: 12px;
    bottom: 0;
    content: '';
    left: 0;
    position: absolute;
    right: 0;
  }
}
