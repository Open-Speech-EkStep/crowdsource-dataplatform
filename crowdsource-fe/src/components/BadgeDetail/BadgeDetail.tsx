import { useEffect, useState } from 'react';

import { useTranslation } from 'next-i18next';
import { useRouter } from 'next/router';
import Tab from 'react-bootstrap/Tab';
import Tabs from 'react-bootstrap/Tabs';

import InitiativeBadgeDetail from 'components/InitiativeBadgeDetail';
import LanguageDropDown from 'components/LanguageDropDown';
import {
  INITIATIVES_MAPPING,
  INITIATIVES_MEDIA_MAPPING,
  INITIATIVES_REVERSE_MEDIA_MAPPING,
} from 'constants/initiativeConstants';
import { MEDALS } from 'constants/medalConstants';
import { sourceConstants } from 'constants/pageRouteConstants';
import routePaths from 'constants/routePaths';
import type { Initiative } from 'types/Initiatives';
import type { InitiativeType } from 'types/InitiativeType';
import type { MedalsType } from 'types/MedalsType';
import type { SourceType } from 'types/SourceType';

import styles from './BadgeDetail.module.scss';

const BadgeDetail = () => {
  const { t } = useTranslation();
  const router = useRouter();
  const { locale: currentLocale } = useRouter();
  const { language, initiative, source, badge } = router.query;

  const [defaultLanguage, setLanguage] = useState((language as string) || 'English');
  const [defaultInitiative, setDefaultInitiative] = useState<Initiative>(
    INITIATIVES_REVERSE_MEDIA_MAPPING[initiative as InitiativeType] || INITIATIVES_MAPPING.suno
  );
  const defaultAction: SourceType = (source as SourceType) || sourceConstants.contribute;
  const defaultBadge = (badge as MedalsType) || MEDALS[0];

  const setRoutingParam = (
    initiative: InitiativeType,
    language: string,
    source: SourceType,
    badge: MedalsType
  ) => {
    router.push(
      `/${currentLocale}${routePaths.badges}?initiative=${initiative}&language=${language}&source=${source}&badge=${badge}`,
      undefined,
      {
        locale: currentLocale,
      }
    );
  };

  function handleTabClick(tab: string | null, event: any) {
    event.target.scrollIntoView({
      behavior: 'smooth',
      block: 'center',
      inline: 'center',
    });
    setDefaultInitiative(tab as Initiative);
    setRoutingParam(
      INITIATIVES_MEDIA_MAPPING[tab as Initiative],
      defaultLanguage,
      defaultAction,
      defaultBadge
    );
  }

  useEffect(() => {
    setRoutingParam(
      INITIATIVES_MEDIA_MAPPING[defaultInitiative],
      defaultLanguage,
      defaultAction,
      defaultBadge
    );
  }, []);

  return (
    <div data-testid="BadgeDetail">
      <header>
        <h2 className="mb-5 mb-md-8">{t('badgesDetailPageHeading')}</h2>
        <LanguageDropDown
          selectedLanguage={defaultLanguage as string}
          updateSelectedLanguage={selectedLanguage => {
            setRoutingParam(
              INITIATIVES_MEDIA_MAPPING[defaultInitiative],
              selectedLanguage,
              defaultAction,
              defaultBadge
            );
            setLanguage(selectedLanguage);
          }}
        />
      </header>
      <div className="font-family-rowdies mt-7 mt-md-9">
        <Tabs
          activeKey={defaultInitiative}
          id="uncontrolled-tab-example"
          className={`${styles.tabs} d-flex align-items-center border-bottom border-1 border-primary-60 hide-scrollbar`}
          onSelect={handleTabClick}
        >
          {Object.values(INITIATIVES_MAPPING).map((initiativeName: string) => (
            <Tab
              key={initiativeName}
              eventKey={initiativeName}
              title={`${t(initiativeName)} ${t('india')}`}
              tabClassName={`${styles.tabLink} position-relative h-100 display-3 px-6 mx-md-0`}
            ></Tab>
          ))}
        </Tabs>
        <div className="py-5">
          <InitiativeBadgeDetail
            initiative={defaultInitiative}
            language={defaultLanguage as string}
            action={defaultAction as SourceType}
            badge={defaultBadge as MedalsType}
            onSelectSourceType={(type: SourceType) =>
              setRoutingParam(
                INITIATIVES_MEDIA_MAPPING[defaultInitiative],
                defaultLanguage,
                type,
                defaultBadge
              )
            }
            onSelectBadgeType={(badge: MedalsType) =>
              setRoutingParam(
                INITIATIVES_MEDIA_MAPPING[defaultInitiative],
                defaultLanguage,
                defaultAction,
                badge
              )
            }
          />
        </div>
      </div>
      <div className="display-4">
        <p className="m-0">{t('badgeInfoText1')}</p>
        <p className="m-0">{t('badgeInfoText2')}</p>
      </div>
    </div>
  );
};

export default BadgeDetail;
