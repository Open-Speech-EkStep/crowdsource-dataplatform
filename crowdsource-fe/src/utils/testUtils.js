import { render } from '@testing-library/react';

const Wrapper = ({ children }) => children;

const customRender = (ui, options) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { renderHook } from '@testing-library/react-hooks';

// override render method
export { customRender as render };
