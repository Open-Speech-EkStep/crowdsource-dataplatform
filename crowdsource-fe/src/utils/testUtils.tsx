/* eslint-disable jest/no-export */
import type { ReactNode, ReactElement, ComponentType } from 'react';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render, act } from '@testing-library/react';
import { axe } from 'jest-axe';
import puppeteer from 'puppeteer';
import type { Browser } from 'puppeteer';

const Wrapper: ComponentType = ({ children }: { children?: ReactNode }) => children as ReactElement;

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { renderHook } from '@testing-library/react-hooks';

// override render method
export { customRender as render };

export const verifyAxeTest = ({ container }: { container: RenderResult['container'] }) => {
  it('should not fail an axe audit', async () => {
    let axeResult;

    await act(async () => {
      axeResult = await axe(container);
    });

    expect(axeResult).toHaveNoViolations();
  });
};

export const verifyVRTest = (title: string, url: string) => {
  describe(`${title} VR test`, () => {
    let browser: Browser;

    beforeAll(async () => {
      browser = await puppeteer.launch({
        headless: true,
        devtools: false,
        ignoreHTTPSErrors: true,
        args: ['--start-fullscreen'],
      });
    });

    it('match snapshot', async () => {
      const page = await browser.newPage();
      await page.setViewport({ width: 1920, height: 1080 });

      await page.goto(url, {
        waitUntil: 'networkidle2',
      });

      await page.addStyleTag({
        content: `
      *,
      *::after,
      *::before {
          transition-delay: 0s !important;
          transition-duration: 0s !important;
          animation-delay: -0.0001s !important;
          animation-duration: 0s !important;
          animation-play-state: paused !important;
          caret-color: transparent !important;
      }
      `,
      });

      const image = await page.screenshot({
        fullPage: true,
      });

      expect(image).toMatchImageSnapshot();
    });

    afterAll(async () => {
      await browser.close();
    });
  });
};
