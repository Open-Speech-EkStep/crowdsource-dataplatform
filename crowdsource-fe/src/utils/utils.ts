import { jsPDF } from 'jspdf';
import { i18n } from 'next-i18next';

import { ErrorStatusCode } from 'constants/errorStatusCode';
import { INITIATIVES_MAPPING } from 'constants/initiativeConstants';
import { KEYBOARD_ERROR, LANGUAGE_UNICODE, OTHER_LANGUAGE_UNICODE } from 'constants/Keyboard';
import type { InitiativeType } from 'types/InitiativeType';

import apiPaths from '../constants/apiPaths';

var platform = require('platform');

export const convertIntoHrsFormat = (data: any, isSeconds = true) => {
  const hours = Math.floor(data / 3600);
  const remainingAfterHours = data % 3600;
  const minutes = Math.floor(remainingAfterHours / 60);
  const seconds = Math.round(remainingAfterHours % 60);
  if (isSeconds) {
    return { hours, minutes, seconds };
  } else {
    return { hours, minutes };
  }
};

export const roundOffValue = (average: any, precision: number) => {
  const formattedAverage = average.toFixed(precision);
  return formattedAverage;
};

export const capitalizeFirstLetter = (strValue: string) => {
  return strValue?.charAt(0).toUpperCase() + strValue?.slice(1);
};

const translateText = (text: string) => i18n?.t(text);

export const formatTime = (hours: number, minutes = 0, seconds = 0) => {
  const hrStr = translateText('hours1');
  const minStr = translateText('minutes1');
  const secStr = translateText('seconds1');
  let result = '';
  if (hours > 0) {
    result += `${hours} ${hrStr} `;
  }
  if (minutes > 0) {
    result += `${minutes} ${minStr} `;
  }
  if (hours === 0 && minutes === 0 && seconds > 0) {
    result += `${seconds} ${secStr} `;
  }

  if (hours === 0 && minutes === 0 && seconds === 0) {
    result += `0 ${secStr} `;
  }

  return result.substr(0, result.length - 1);
};

export const convertTimeFormat = (value: any) => {
  const { hours, minutes, seconds } = convertIntoHrsFormat(Number(value) * 60 * 60);
  const data = formatTime(hours, minutes, seconds);
  return data;
};

const roundDuration = (value: number) => {
  return Math.round(value * 100) / 100;
};

export const getHoursValue = (value: any) => {
  return roundDuration(value);
};

export const getMinutesValue = (value: any) => {
  return roundDuration(value * 60);
};

export const getHoursText = (value: any) => {
  return `${getHoursValue(value)} ${i18n?.t('hours2')}`;
};

export const getMinutesText = (value: any) => {
  return `${getMinutesValue(value)} ${i18n?.t('minutes2')}`;
};

export const isSunoOrBoloInitiative = (value: InitiativeType) => {
  const boloOrSuno = ['asr', 'text'];
  return boloOrSuno.includes(value);
};

export const isBoloInitiative = (value: InitiativeType) => {
  const boloOrSuno = ['text'];
  return boloOrSuno.includes(value);
};

export const findInputError = (text: string, currentModule: string, language: string) => {
  if (!text.length) return;
  const newText = text.replace(/\s/g, ''); //read input value, and remove "space" by replace \s
  let error: any = KEYBOARD_ERROR.language;
  const isSunoModule = currentModule === INITIATIVES_MAPPING.suno;
  const specialSymbols = /[\u0021-\u002F\u003A-\u0040\u005B-\u0060\u007B-\u007E\u0964-\u0965]/;
  if (currentModule === INITIATIVES_MAPPING.suno && specialSymbols.test(newText) === true) {
    return KEYBOARD_ERROR.symbol;
  }
  Object.entries(isSunoModule ? LANGUAGE_UNICODE : OTHER_LANGUAGE_UNICODE).forEach(([key, value]) => {
    if (value.test(newText)) {
      //Check Unicode to see which one is true
      if (key.toLowerCase() === language.toLowerCase()) {
        error = KEYBOARD_ERROR.noError;
      }
    }
  });
  return error;
};

export const fetchLocationInfo = async () => {
  const result = await fetch('https://www.cloudflare.com/cdn-cgi/trace')
    .then(res => res.text())
    .then(async ipAddressText => {
      const dataArray = ipAddressText.split('\n');
      let ipAddress = '';
      for (let ind in dataArray) {
        if (dataArray[ind].startsWith('ip=')) {
          ipAddress = dataArray[ind].replace('ip=', '');
          break;
        }
      }
      if (ipAddress.length !== 0) {
        const data = await fetch(`${apiPaths.locationInfo}?ip=${ipAddress}`);
        return data && data.json();
      } else {
        return new Promise((resolve, reject) => {
          reject('Ip Address not available');
        });
      }
    });

  return result;
};

export const getDeviceInfo = () => {
  const os = platform.os;
  return os.family + ' ' + os.version;
};

export const getBrowserInfo = () => {
  return platform.name + ' ' + platform.version;
};

export const downloadBadge = (
  locale: string | undefined,
  initiative: string,
  source: string,
  badgeType: string,
  badgeId: string
) => {
  const pdf = new jsPDF();
  const img = new Image();
  img.src = `/images/${locale}/badges/${locale}_${initiative}_${badgeType}_${source}.png`;
  img.crossOrigin = 'Anonymous';
  img.onload = function () {
    pdf.addImage(img, 50, 10, 105, 130);
    pdf.save(`${badgeType}-badge.pdf`);
  };

  pdf.text(`Badge Id : ${badgeId}`, 36, 190);
};

export const groupBy = function (list: any, keyName: any) {
  if (!list || list.length === 0 || !list[0][keyName]) {
    return [];
  }
  return list.reduce(function (pair: any, key: any) {
    (pair[key[keyName]] = pair[key[keyName]] || []).push(key);
    return pair;
  }, {});
};

/* istanbul ignore next */
export const visualize = (visualizer: any, analyser: any) => {
  const canvasCtx = visualizer.getContext('2d');
  const bufferLength = analyser.frequencyBinCount;
  const dataArray = new Uint8Array(bufferLength);
  const WIDTH = visualizer.width;
  const HEIGHT = visualizer.height;
  // TODO do we need to limit the number of time visualize refreshes per second
  // so that it can run on Android processors without causing audio to drop?
  function draw() {
    // this is more efficient than calling with processor.onaudioprocess
    // and sending floatarray with each call...
    requestAnimationFrame(draw);
    analyser.getByteTimeDomainData(dataArray);
    // canvasCtx.fillStyle = 'rgb(255, 255, 255, 0)';
    // canvasCtx.fillRect(0, 0, WIDTH, HEIGHT);
    canvasCtx.clearRect(0, 0, WIDTH, HEIGHT);
    canvasCtx.lineWidth = 2;
    canvasCtx.strokeStyle = 'rgb(0,123,255)';
    canvasCtx.beginPath();
    const sliceWidth = (WIDTH * 1.0) / bufferLength;
    let x = 0;
    for (let i = 0; i < bufferLength; i++) {
      let v = dataArray[i] / 128.0; // uint8
      let y = (v * HEIGHT) / 2; // uint8
      if (i === 0) {
        canvasCtx.moveTo(x, y);
      } else {
        canvasCtx.lineTo(x, y);
      }
      x += sliceWidth;
    }
    canvasCtx.lineTo(visualizer.width, visualizer.height / 2);
    canvasCtx.stroke();
  }

  draw();
};

export const getErrorMsg = (error: any) => {
  if (error && error.status === ErrorStatusCode.TOO_MANY_REQUEST) {
    return 'multipleRequestApiError';
  } else {
    return 'apiFailureError';
  }
};
