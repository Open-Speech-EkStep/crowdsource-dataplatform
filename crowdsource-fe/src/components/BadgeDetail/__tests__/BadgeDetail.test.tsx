import { when } from 'jest-when';

import { render, screen, userEvent } from 'utils/testUtils';

import BadgeDetail from '../BadgeDetail';

describe('BadgeDetail', () => {
  const setup = () => {
    when(localStorage.getItem)
      .calledWith('contributionLanguage')
      .mockImplementation(() => 'English');

    when(sessionStorage.getItem)
      .calledWith('prevPath')
      .mockImplementation(() => '/suno-india');

    return render(<BadgeDetail />);
  };

  it('should render the component and matches it against stored snapshot', () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render badges in "Assamese" language when Assamese is chosen from language dropdown', async () => {
    setup();

    userEvent.selectOptions(screen.getByTestId('SelectDropDownLanguage'), 'অসমীয়া');

    expect(screen.getByRole('combobox', { name: 'Your Language' })).toHaveValue('Assamese');
  });

  it('should render badge detail for selected initiative', async () => {
    let scrollIntoViewMock = jest.fn();
    window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;

    setup();
    userEvent.click(screen.getByRole('tab', { name: 'bolo india' }));

    expect(scrollIntoViewMock).toBeCalled();
    userEvent.click(screen.getByRole('tab', { name: 'bolo india' }));
    expect(screen.getByRole('tab', { name: 'bolo india' })).toHaveClass('active');
  });

  it('should activate validate badges when validate radio button is selected', async () => {
    await setup();
    expect(screen.getByTestId('action2')).not.toBeChecked();
    userEvent.click(screen.getByTestId('action2'));
    expect(screen.getByTestId('action2')).toBeChecked();
    expect(screen.getByTestId('action1')).not.toBeChecked();
  });
});
