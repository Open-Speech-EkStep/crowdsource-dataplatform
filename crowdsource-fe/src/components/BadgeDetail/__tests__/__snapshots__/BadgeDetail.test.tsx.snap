// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeDetail should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="BadgeDetail"
  >
    <header>
      <h2
        class="mb-5 mb-md-8"
      >
        badgesDetailPageHeading
      </h2>
      <div
        class="row"
      >
        <div
          class="col"
          data-testid="LanguageDropDown"
        >
          <div
            class="contributionLanguage"
          >
            <div
              class="d-flex flex-column flex-md-row align-items-md-center"
            >
              <label
                class="label mb-0 me-md-2 form-label"
                for="languageDropDown"
              >
                selectDropDownLanguagePrompt:
              </label>
              <select
                aria-label="Your Language"
                class="languageDropdown mt-1 mt-md-0 form-select"
                data-testid="SelectDropDownLanguage"
                id="languageDropDown"
                name="languageDropDown"
              >
                <option
                  value="Assamese"
                >
                  অসমীয়া
                </option>
                <option
                  value="Bengali"
                >
                  বাংলা
                </option>
                <option
                  value="English"
                >
                  English
                </option>
                <option
                  value="Gujarati"
                >
                  ગુજરાતી
                </option>
                <option
                  value="Hindi"
                >
                  हिंदी
                </option>
                <option
                  value="Kannada"
                >
                  ಕನ್ನಡ
                </option>
                <option
                  value="Malayalam"
                >
                  മലയാളം
                </option>
                <option
                  value="Marathi"
                >
                  मराठी
                </option>
                <option
                  value="Odia"
                >
                  ଓଡିଆ
                </option>
                <option
                  value="Punjabi"
                >
                  ਪੰਜਾਬੀ
                </option>
                <option
                  value="Tamil"
                >
                  தமிழ்
                </option>
                <option
                  value="Telugu"
                >
                  తెలుగు
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="font-family-rowdies mt-7 mt-md-9"
    >
      <ul
        class="tabs d-flex align-items-center border-bottom border-1 border-primary-60 hide-scrollbar nav nav-tabs"
        role="tablist"
      >
        <li
          class="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-tts"
            aria-selected="true"
            class="tabLink position-relative h-100 display-3 px-6 mx-md-0 nav-link active"
            data-rr-ui-event-key="tts"
            id="uncontrolled-tab-example-tab-tts"
            role="tab"
            type="button"
          >
            tts initiativeTextSuffix
          </button>
        </li>
        <li
          class="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-asr"
            class="tabLink position-relative h-100 display-3 px-6 mx-md-0 nav-link"
            data-rr-ui-event-key="asr"
            id="uncontrolled-tab-example-tab-asr"
            role="tab"
            tabindex="-1"
            type="button"
          >
            asr initiativeTextSuffix
          </button>
        </li>
        <li
          class="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-translation"
            class="tabLink position-relative h-100 display-3 px-6 mx-md-0 nav-link"
            data-rr-ui-event-key="translation"
            id="uncontrolled-tab-example-tab-translation"
            role="tab"
            tabindex="-1"
            type="button"
          >
            translation initiativeTextSuffix
          </button>
        </li>
        <li
          class="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-ocr"
            class="tabLink position-relative h-100 display-3 px-6 mx-md-0 nav-link"
            data-rr-ui-event-key="ocr"
            id="uncontrolled-tab-example-tab-ocr"
            role="tab"
            tabindex="-1"
            type="button"
          >
            ocr initiativeTextSuffix
          </button>
        </li>
      </ul>
      <div
        class="tab-content"
      >
        <div
          aria-labelledby="uncontrolled-tab-example-tab-tts"
          class="tab-pane active"
          id="uncontrolled-tab-example-tabpane-tts"
        />
        <div
          aria-labelledby="uncontrolled-tab-example-tab-asr"
          class="tab-pane"
          id="uncontrolled-tab-example-tabpane-asr"
        />
        <div
          aria-labelledby="uncontrolled-tab-example-tab-translation"
          class="tab-pane"
          id="uncontrolled-tab-example-tabpane-translation"
        />
        <div
          aria-labelledby="uncontrolled-tab-example-tab-ocr"
          class="tab-pane"
          id="uncontrolled-tab-example-tabpane-ocr"
        />
      </div>
      <div
        class="py-5"
      >
        <div
          data-testid="InitiativeBadgeDetail"
        >
          <div
            class="py-5 row"
            data-testid="badgeLevelRow"
          >
            <div
              class="d-flex align-items-center fw-light display-3 col-md-3"
            >
              selectBadgeLevelPrompt
            </div>
            <div
              class="d-flex justify-content-between justify-content-lg-start mt-5 mt-md-0 col-md-9"
            >
              <div
                class="medal"
              >
                <div
                  class="root active d-flex flex-column align-items-center text-center py-2 py-md-3"
                  data-testid="en-tts-bronze-contribute-medal"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="medalImg d-flex"
                  >
                    <span
                      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                    >
                      <span
                        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                      >
                        <img
                          alt=""
                          aria-hidden="true"
                          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2756%27%20height=%2772%27/%3e"
                          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                        />
                      </span>
                      <img
                        alt="Medal"
                        data-nimg="intrinsic"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                      />
                      <noscript />
                    </span>
                  </div>
                  <span
                    class="display-5 mt-1 fw-light text-capitalize text-break"
                  >
                    Bronze
                  </span>
                </div>
              </div>
              <div
                class="medal"
              >
                <div
                  class="root  d-flex flex-column align-items-center text-center py-2 py-md-3"
                  data-testid="en-tts-silver-contribute-medal"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="medalImg d-flex"
                  >
                    <span
                      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                    >
                      <span
                        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                      >
                        <img
                          alt=""
                          aria-hidden="true"
                          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2756%27%20height=%2772%27/%3e"
                          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                        />
                      </span>
                      <img
                        alt="Medal"
                        data-nimg="intrinsic"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                      />
                      <noscript />
                    </span>
                  </div>
                  <span
                    class="display-5 mt-1 fw-light text-capitalize text-break"
                  >
                    Silver
                  </span>
                </div>
              </div>
              <div
                class="medal"
              >
                <div
                  class="root  d-flex flex-column align-items-center text-center py-2 py-md-3"
                  data-testid="en-tts-gold-contribute-medal"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="medalImg d-flex"
                  >
                    <span
                      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                    >
                      <span
                        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                      >
                        <img
                          alt=""
                          aria-hidden="true"
                          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2756%27%20height=%2772%27/%3e"
                          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                        />
                      </span>
                      <img
                        alt="Medal"
                        data-nimg="intrinsic"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                      />
                      <noscript />
                    </span>
                  </div>
                  <span
                    class="display-5 mt-1 fw-light text-capitalize text-break"
                  >
                    Gold
                  </span>
                </div>
              </div>
              <div
                class="medal"
              >
                <div
                  class="root  d-flex flex-column align-items-center text-center py-2 py-md-3"
                  data-testid="en-tts-platinum-contribute-medal"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="medalImg d-flex"
                  >
                    <span
                      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                    >
                      <span
                        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                      >
                        <img
                          alt=""
                          aria-hidden="true"
                          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2756%27%20height=%2772%27/%3e"
                          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                        />
                      </span>
                      <img
                        alt="Medal"
                        data-nimg="intrinsic"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                      />
                      <noscript />
                    </span>
                  </div>
                  <span
                    class="display-5 mt-1 fw-light text-capitalize text-break"
                  >
                    Platinum
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="py-5 row"
            data-testid="participationRow"
          >
            <div
              class="d-flex align-items-center fw-light display-3 col-md-3"
            >
              participation
            </div>
            <div
              class="d-flex justify-content-between justify-content-md-start mt-5 mt-md-0 col-md-9"
            >
              <div
                class="fw-light display-3"
              >
                <div
                  class="me-9 form-check form-check-inline"
                >
                  <input
                    checked=""
                    class="form-check-input"
                    data-testid="action1"
                    id="action1"
                    name="action"
                    type="radio"
                    value="Contribution"
                  />
                  <label
                    class="form-check-label"
                    for="action1"
                    title=""
                  >
                    contribution
                  </label>
                </div>
                <div
                  class="form-check form-check-inline"
                >
                  <input
                    class="form-check-input"
                    data-testid="action2"
                    id="action2"
                    name="action"
                    type="radio"
                    value="Validation"
                  />
                  <label
                    class="form-check-label"
                    for="action2"
                    title=""
                  >
                    validation
                  </label>
                </div>
              </div>
            </div>
          </div>
          <section
            class="py-5"
          >
            <div
              class="root rounded-20"
              data-testid="badgeDescriptionRow"
            >
              <div
                class="rounded-20 bg-light p-5"
              >
                <div
                  class="row"
                >
                  <div
                    class="d-flex align-items-center justify-content-center col-md-4 col-12 order-md-2 order-1"
                  >
                    <div
                      class="medalImg"
                    >
                      <span
                        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                      >
                        <span
                          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                        >
                          <img
                            alt=""
                            aria-hidden="true"
                            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27250%27%20height=%27320%27/%3e"
                            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                          />
                        </span>
                        <img
                          alt="Bronze Badge English"
                          data-nimg="intrinsic"
                          decoding="async"
                          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                        />
                        <noscript />
                      </span>
                    </div>
                  </div>
                  <div
                    class="d-flex align-items-center justify-content-center text-center text-md-start mt-5 mt-md-0 col-md-8 col-12 order-md-1 order-2"
                  >
                    <div
                      class="heading"
                    >
                      <h2>
                        badgeDetailText
                      </h2>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      class="display-4"
    >
      <p
        class="m-0"
      >
        badgeInfoText1
      </p>
      <p
        class="m-0"
      >
        badgeInfoText2
      </p>
    </div>
  </div>
</DocumentFragment>
`;
